### ISSUE TITLE
Should we have 'minishift install-cli' to download, make it executable 'oc' binary?

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#88]
Hi,

As a part of better user experience, can we have 'oc' binary installation as part of minishift CLI command?
It can be:

```
minishift install-oc (or any name)
# instruction to export binary path
# or eval command for $(minishift install-oc)
```

Right now, user has to find the 'oc' binary by its own.

**Comment by jimmidyson** [comment]
I'd like to have some packages to do this & have dependencies set up appropriately, e.g. brew for osx, rpm for fedora, etc.

**Comment by budhrg** [comment]
@jimmidyson ,

how about this way?
1. try the get the oc binary url from here https://github.com/openshift/origin/releases/tag/v1.2.1 (parsing the page and get the full url like https://github.com/openshift/origin/releases/download/v1.2.1/openshift-origin-client-tools-v1.2.1-5e723f6-linux-64bit.tar.gz
2. download using curl (as it is available in most of platform)

This way we need not to setup dependencies too.

**Comment by jimmidyson** [comment]
Yeah it's easy to get the binary from releases page. Don't use curl but rather use go's `http.Get` & stream to desired file location (well to tmp then rename to be as atomic as possible).

**Comment by budhrg** [comment]
@jimmidyson , I think we might give default location as current folder where user is running minishft and expose option like --path to let user decide its desired file location.

**Comment by jimmidyson** [comment]
As stuff like oc needs to be in PATH might be best to use the same dir as minishift is in rather than the current folder?

**Comment by budhrg** [comment]
@jimmidyson what would works too :+1:

**Comment by jimmidyson** [comment]
As long as it is clear to the user what is happening & they are prompted to download binaries rather than being hidden from the user i'm happy.

**Comment by budhrg** [comment]
@jimmidyson 

> prompted to download binaries rather than being hidden from the user

Does it mean it is part of "minishift start" command?
I was thinking this workflow with it.

- `minishift start` which gives message like do 'oc login -....' (we can append message as do 'minishift install-oc' if user don't have oc binary)
- `minishift install-oc` (which download the binary into $PATH where minishift lies)
- `oc login --user ....`
- User performs other operations.

I also do a value in setting this as part of `minishift start` command which invokes `minishift install-oc` or may be integrated in 'start' command itself.

What do you think?

**Comment by jimmidyson** [comment]
Yeah sounds good. I like the idea of `minishift start` invoking `install-oc` as part of workflow.

**Comment by budhrg** [comment]
Cool then. Will work on it @jimmidyson :+1:

**Comment by hferentschik** [comment]
This will be addressed by issue #142
